### PR TITLE
[9.x] Add Assertion for JsonResponse in payload

### PR DESCRIPTION
**Pull request by CharlesBilbo**
add some basic assertion methods that can be used to assert JSON response containing an API response structure. 

carbon (1)

### CODE REVIEW

**Comment by driesvints** [comment]
Please remove the scalar typehints and update the docblocks to match the ones from the rest of the framework.

File: src/Illuminate/Testing/TestResponse.php [hunk 1]
```diff
@@ -970,6 +971,58 @@ public function assertJsonMissingValidationErrors($keys = null, $responseKey = '
         return $this;
     }
 
+    /**
+     * Assert that the response contains the structure of the JsonResponse.
+     *
+     * @param  JsonResource  $resource
```
Comment (driesvints) [inline src/Illuminate/Testing/TestResponse.php:977]: We use FQN everywhere.

```suggestion
     * @param  \Illuminate\Http\Resources\Json\JsonResource  $resource
```

Comment (CharlesBilbo) [reply src/Illuminate/Testing/TestResponse.php]: Updated, thanks!

**Comment by taylorotwell** [comment]
Thanks for your pull request to Laravel!

Unfortunately, I'm going to delay merging this code for now. To preserve our ability to adequately maintain the framework, we need to be very careful regarding the amount of code we include. 

If possible, please consider releasing your code as a package so that the community can still take advantage of your contributions! 

If you feel absolutely certain that this code corrects a bug in the framework, please "**@**" mention me in a follow-up comment with further explanation so that GitHub will send me a notification of your response.

**Pull request closed without merging**